File: S2P2_herencia/Herencia/Python/Mamifero.py
class Mamifero: #Clase padre - clase base - superclase
    def __init__(self, n="", p=0.0):
        self.__nombre = n #privado
        self._peso = p #protegido

    @property #getter
    def nombre(self):
        return self.__nombre

    @nombre.setter #setter
    def nombre(self, n):
        self.__nombre = n

    def __str__(self): #equivalente a toString
        return f"Mamifero\n [ {self.nombre},{self._peso} kg]"


class Gato(Mamifero):

    def __init__(self, n="", p=0.0, b=0):
        super().__init__(n, p)
        self.__nBigotes = b
        
    def __str__(self):
        return f"{super().__str__()} [Bigotes:{self.__nBigotes}]"


class Vaca(Mamifero):

    def __init__(self, n="", p=0.0, l=0.0):
        super().__init__(n, p)
        self.__litrosDeLeche = l
    
    def calcularCantidadComida(self):
        print(f"{self._peso * 0.03} kg de alimento")

    def __str__(self):
        return f"{super().__str__()} [Litros de leche:{self.__litrosDeLeche}]"


class Ballena(Mamifero):

    def __init__(self, n="", p=0.0):
        super().__init__(n, p)

    def __str__(self):
        return f"{super().__str__()}"

File: 1POO/Semestre 1 - 2/S2P2_herencia/C#/Mamifero/Mamifero.cs
using System;

class Mamifero{
    private string nombre;
    protected double peso;

    public Mamifero(string n, double p)
    {
        this.nombre = n;
        this.peso = p;
    }

    public string Nombre
    {
        get { return this.nombre; }
        set { this.nombre = value; }
    }

    public override string ToString()
    {
        return "Mamifero\n [" + nombre + "," + peso + " kg ]";
    }
}

//clase gato : herencia
class Gato : Mamifero 
{
    private int nBigotes;
//base es para representar la herencia o lo que se hereda, en java es super
    public Gato(string n, double p, int b) : base(n, p)
    {
        this.nBigotes = b;
    }

    public override string ToString()
    {
        return base.ToString()+"[Bigotes:" + nBigotes + "]";
    }
}

class Vaca : Mamifero
{
    private double litrosDeLeche;

    public Vaca(String n, double p, double l) : base(n, p)
    {
        this.litrosDeLeche = l;
    }

    public void calcularCantidadComida()
    {
        Console.WriteLine(this.peso * 0.03 + " kg de alimento");
    }

    public override string ToString()
    {
        return base.ToString() + "[Litros de leche:" + litrosDeLeche + "]";
    }
}

class Ballena : Mamifero{

    public Ballena(string n, double p) : base(n, p)
    {
    }

    public override string ToString()
    {
        return base.ToString();
    }
}

